hera_mc/watch_dog.py

```diff
@@ -86,25 +91,34 @@ def node_verdict(age_out=3800, To=None, testing=False, redishost='redishost'):
             node_info.setdefault(node, {})
             hw = key.split(':')[3]
             node_info[node][hw] = r.hgetall(key)
-    param = r.hgetall('verdict')
-    if not param:
-        return {}
-    msg_header = "Node verdict:  mode '{}', timeout {}, time {}\n".format(
-        param['mode'], param['timeout'], int(float(param['time'])))
-    msg = '{}'.format(msg_header)
-    if int(time.time() - float(param['time'])) < age_out:
-        for node in sorted(node_info.keys()):
-            s = "Node {} is{}valid\n".format(
-                node, ' ' if node_info[node]['valid'] else ' not ')
-            for hw, val in node_info[node].items():
-                if hw != 'valid':
-                    s += "\t{}: {}\n".format(hw, val)
-            s += '\n'
-            msg += s
-    node_info['param'] = param
-    if msg != msg_header:
-        return send_email(msg_header.splitlines()[0], msg, To, skip_send=testing)
-    return node_info
+    node_list = sorted(node_info.keys())
+    node_info['param'] = r.hgetall('verdict')
+    this_email = None
+    if node_info['param']:
+        this_time = float(node_info['param']['time'])
+        subject = "Node verdict:  mode '{}', timeout {}, time {}".format(
+            node_info['param']['mode'],
+            node_info['param']['timeout'],
+            int(this_time))
+        msg = ''
+        if int(time.time() - this_time) < age_out:
+            for node in node_list:
+                s = "Node {} is{}valid\n".format(
+                    node, ' ' if node_info[node]['valid'] else ' not ')
+                for hw, val in node_info[node].items():
+                    if hw != 'valid':
+                        s += "\t{}: {}\n".format(hw, val)
+                s += '\n'
+                msg += s
+        if len(msg):
+            msg = subject + '\n\n' + msg
+            this_email = send_email(subject, msg, To, skip_send=testing)
+    if return_as == 'node_info':
+        return node_info
+    elif return_as == 'email':
+        return this_email
+    else:
+        raise ValueError("Must be returned as 'node_info' or 'email'")
```

Comment:
I like that you're checking that this, but I'd do it earlier so it's a fast fail rather than only failing after it did all the work.

Reply:
Done